### PR TITLE
feature/SCKAN-128 implement statement ingestion command

### PR DESCRIPTION
**Pull request by Anto8819**
SCKAN-127 neurons ingestion WIP

SCKAN-128 check if entities already exist in composer db

SCKAN-128 create artifact sentence for each of the statements to be ingested

SCKAN-128 clean unused function

SCKAN-128 fix function name

SCKAN-128 add reference_uri field

SCKAN-128 fix create reference_uri if not provided

SCKAN-128 add functiona circuit and projection phenotype fields

SCKAN-128 directory structure and create sentence changes

SCKAN-128 add alert note type and prepare data for ingestion WIP

SCKAN-128 validate and create statements

SCKAN-128 add many to many fields to new statements

SCKAN-128 add sex and spec validation and  warning for skipped phenotypes

SCKAN-128 add direct sentence transition from OPEN to COMPOSE_NOW if user is system

SCKAN-128 fix multiple provenances and add TODOs

### CODE REVIEW

File: backend/composer/services/cs_ingestion/cs_ingestion_services.py [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+            provenances = (Provenance(connectivity_statement = connectivity_statement, uri=provenance) for provenance in provenances_list)
+            Provenance.objects.bulk_create(provenances)
+            # only 5 statements have a note_alert but they were all filtered out since at least one of their anatomical entities was not found in composer db
+            if has_prop(statement[NOTE_ALERT]):   
+                Note.objects.create(connectivity_statement = connectivity_statement, user=User.objects.get(username="system"), type=NoteType.ALERT, note=statement[NOTE_ALERT][0])
+            connectivity_statement.save()
+
+            # transition to compose_now
+            do_transition_to_compose_now(sentence)
+
```
Comment (zsinnema) [inline backend/composer/services/cs_ingestion/cs_ingestion_services.py:233]: @Anto8819 do we also need to transition the statement to a new state?

File: backend/composer/services/cs_ingestion/cs_ingestion_services.py [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+            return model.objects.filter(ontology_uri=prop[0])[0]
+        else:
+            return model.objects.get(ontology_uri=prop[0])
+    else:
+        return None
+
+
+
+def map_circuit_type_uri_with_model(uri):
+    circuit_type = {
```
Comment (zsinnema) [inline backend/composer/services/cs_ingestion/cs_ingestion_services.py:152]: @Anto8819 I suggest to make this a CONSTANT and move it up in the document (just below the imports)

File: backend/composer/management/commands/ingest_anatomical_entities.py [hunk 3]
```diff
@@ -35,6 +35,7 @@ def handle(self, *args, **options):
                     quotechar='"',
                 )
                 for row in aereader:
+                    import pdb; pdb.set_trace()
```
Comment (zsinnema) [inline backend/composer/management/commands/ingest_anatomical_entities.py:38]: @Anto8819 please remove this debug line

**Review by zsinnema — CHANGES_REQUESTED** [review]
@Anto8819 looks great, just some small stuff and a question if we should move the statement into a different state?

File: backend/composer/services/cs_ingestion/cs_ingestion_services.py (hunk 1 above)
Comment (Anto8819) [reply backend/composer/services/cs_ingestion/cs_ingestion_services.py:233]: @zsinnema you mean to the EXPORTED state, right? yep, I forgot that one, I'll add it

Comment (zsinnema) [reply backend/composer/services/cs_ingestion/cs_ingestion_services.py:233]: @Anto8819 yep

Comment (Anto8819) [reply backend/composer/services/cs_ingestion/cs_ingestion_services.py]: done

**Comment by Anto8819** [comment]
done @zsinnemal. Also, I updated the anatomical_entities.cvs with the latest provided by the client

**Comment by Anto8819** [comment]
done @zsinnema . Also, I updated the anatomical_entities.csv with the latest provided by the client

File: backend/composer/services/cs_ingestion/cs_ingestion_services.py [hunk 4]
```diff
@@ -0,0 +1,251 @@
...
+    if CSState.EXPORTED in available_transitions:
+        # after the sentence and statement transitioned to compose_now, we need to update the statement state to exported 
+        cs = ConnectivityStatementService(statement).do_transition(
+            CSState.EXPORTED, system_user
+        )
+        cs.save()
+
+def do_state_transitions(sentence: Sentence):
+   s = do_transition_to_compose_now(sentence)
+   do_transition_to_exported(s.connectivitystatement_set.first())
```
[the remark this answers is not in the document]
Comment (Anto8819) [reply backend/composer/services/cs_ingestion/cs_ingestion_services.py]: done

**Review by zsinnema — APPROVED** [review]

**Pull request merged**